be/src/vec/exec/vbroker_scan_node.h

```diff
@@ -43,15 +54,64 @@ class VBrokerScanNode final : public BrokerScanNode {
     // Close the scanner, and report errors.
     Status close(RuntimeState* state) override;
 
+    // No use
+    Status set_scan_ranges(const std::vector<TScanRangeParams>& scan_ranges) override;
+
 private:
-    Status start_scanners() override;
+    // Write debug string of this into out.
+    void debug_string(int indentation_level, std::stringstream* out) const override;
+
+    // Update process status to one failed status,
+    // NOTE: Must hold the mutex of this scan node
+    bool update_status(const Status& new_status) {
+        if (process_status_.ok()) {
+            process_status_ = new_status;
+            return true;
+        }
+        return false;
+    }
+
+    std::unique_ptr<BaseScanner> create_scanner(const TBrokerScanRange& scan_range,
+                                                ScannerCounter* counter);
+
+    Status start_scanners();
 
     void scanner_worker(int start_idx, int length);
     // Scan one range
     Status scanner_scan(const TBrokerScanRange& scan_range, ScannerCounter* counter);
 
-    std::deque<std::shared_ptr<vectorized::Block>> _block_queue;
-    std::unique_ptr<MutableBlock> _mutable_block;
+    TupleId tuple_id_;
```

Comment:
The member variables here may be repeated to "broker_scan_node.h".

Reply:
This node inherits from scan node and is independent from BrokerScanNode